packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -834,8 +834,21 @@ abstract class ResidentRunner {
       'sksl',
     );
     final Device device = flutterDevices.first.device;
+
+    // Convert android sub-platforms to single target platform.
```

Comment:
Nit: is this conversion just to reduce the number of platform-mismatch warnings? If so, I'd probably prefer to not convert, and print warnings if one tries to use the SkSL bundle from android_x64 on android_arm64.

Reply:
I'm not sure how this would be possible today. When building an apk or appbundle, the tool builds for all ABIs simultaneously and assumes that assets are shared. It would take a significant amount of refactoring to support ABI-specific assets.

Reply:
Are the SkSL files ABI specific? If so this needs a lot more work :(

Reply:
No, they're not ABI specific. I was only thinking about giving them warnings as platform-mismatch may make them less efficient. But there's absolutely no problem of cross-ABI compatibility (as I've used x64 SkSLs on arm).

Reply:
This might be something we could fix long term then, would you file an issue with some more details on the efficiency loss from the current implementation?

Reply:
Here's an experiment on the efficiency loss: https://github.com/flutter/flutter/issues/53607#issuecomment-608587484

I don't think the current implementation has any issues. If we truly want to be 100% effective across different ABIs/devices, we probably have to rely on our medium-term solution "Test-based shader warmup #53609".